File: combat_genetics/src/Bullet.cpp
//
//  Bullet.cpp
//  combat_genetics
//
//  Created by Andrew Wallace on 9/30/16.
//
//

#include "Bullet.hpp"

void Bullet::setup(float x, float y, float angle){
    
    float speed = 4;
    size = 5;
    
    pos.set(x,y);
    vel.x = cos(angle) * speed;
    vel.y = sin(angle) * speed;
    
}

void Bullet::update(){
    pos += vel;
}

void Bullet::draw(){
    ofNoFill();
    ofSetColor(255);
    ofDrawCircle(pos.x, pos.y, size);
}

File: p5js_project/Bullet.js
function Bullet(){

	this.size = 5;
	this.speed = 4;

	this.pos;
	this.vel;

	this.angle;

	this.setup = function(x, y, _angle){
		this.angle = _angle;
		this.pos = createVector(x, y, 0);
		
		this.vel = createVector( cos(this.angle) * this.speed, sin(this.angle) * this.speed, 0 );
	}

	this.update = function(){
		this.pos.x += this.vel.x;
	    this.pos.y += this.vel.y;
	}

	this.draw = function(){
	    noFill();
	    stroke(255);
	    ellipse(this.pos.x, this.pos.y, this.size, this.size);
	}

}